tests/jenkins/run_test_pip_installations.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+set -e
+
+if (( $# < 1 )); then
+    echo ""
+    echo "Usage: $(basename $0) WORKSPACE_PATH"
+    echo ""
+    exit 1
+fi
+
+WORKSPACE=$( echo "$1" | tr '[:upper:]' '[:lower:]' )
+
+PYTHON_VERSIONS=('2.7' '3.4' '3.5' '3.6')
+DEVICES=('pip_cpu' 'pip_cu75' 'pip_cu80')
+
+# build Docker images and test pip installation for each device
+for DEV in "${DEVICES[@]}"; do
+
+    # get Docker binary
+    if [[ "${DEV}" == *"cpu"* ]]; then
+        DOCKER_BINARY="docker"
+    else
+        DOCKER_BINARY="nvidia-docker"
+    fi
+
+    # concatenate the Dockerfile with dependencies into the device file
+    DOCKERFILE="Dockerfile.${DEV}"
+    DOCKERFILE_DEVICE="Dockerfile.in.${DEV}"
+    rm -rf ${DOCKERFILE}
+    cp ${DOCKERFILE_DEVICE} ${DOCKERFILE}
+    cat Dockerfile.pip_dependencies >> ${DOCKERFILE}
+
+    # build Docker image
```

Comment:
1. It can rebuild if we've made changes to the Dockerfile
2. We don't assume that the image exists on the host
3. We don't have to rely on Dockerhub/ECS and push small images that we only use for testing purposes

What's nice about this is that if we've rerun this test in succession, the image built the first time is cached, so we don't spend additional time rebuilding.